### PR TITLE
docs: Add documentation on how to use VSCode to debug systemd via mkosi

### PR DESCRIPTION
**Pull request by daandemeyer**
This is mostly for myself so I don't have to figure this out another time but I assume others might find it useful as well.

### CODE REVIEW

File: docs/HACKING.md [hunk 1]
```diff
@@ -245,3 +245,83 @@ the cached images are initialized (`mkosi -i`).
...
+```
+
+Now that the debugger knows how to connect to our process in the container/VM and we've set up the necessary
+source mappings, go to the "Run and Debug" window and run the "systemd" debug configuration. If everything
+goes well, the debugger should now be attached to the systemd instance running in the container/VM. You can
+attach breakpoints from the editor and enjoy all the other features of VSCode's debugger.
+
+To debug systemd components other than PID 1, set "program" to the full path of the component you want to
+debug and set "processId" to "${command:pickProcess}". Now, when starting the debugger, VSCode will ask you
+the PID of the process you want to debug. Run `systemctl status <component>` in the container to figure out
```
Comment (bluca) [inline docs/HACKING.md]: `systemctl show --property --value MainPID <component>`?

**Review by bluca — APPROVED** [review]
Neat, will try this! One nit

File: docs/HACKING.md [hunk 2]
```diff
@@ -245,3 +245,83 @@ the cached images are initialized (`mkosi -i`).
...
+```
+
+Now that the debugger knows how to connect to our process in the container/VM and we've set up the necessary
+source mappings, go to the "Run and Debug" window and run the "systemd" debug configuration. If everything
+goes well, the debugger should now be attached to the systemd instance running in the container/VM. You can
+attach breakpoints from the editor and enjoy all the other features of VSCode's debugger.
+
+To debug systemd components other than PID 1, set "program" to the full path of the component you want to
+debug and set "processId" to "${command:pickProcess}". Now, when starting the debugger, VSCode will ask you
+the PID of the process you want to debug. Run `systemctl show --property --value MainPID <component>` in the
```
Comment (bluca) [inline docs/HACKING.md]: sorry, I gave you a bad command, --value goes after MainPID, ie: `systemctl show --property MainPID --value <component>`

**Comment by aaomidi** [comment]
I'm having trouble figuring out how to use this guide to debug systemctl, any ideas?

**Comment by DaanDeMeyer** [comment]
This describes debugging daemons. For the CLI tools this wouldn't work. It might be possible to modify it so that it launches the executable for you but I havnen't tried this out yet.

**Comment by aaomidi** [comment]
I did confirm it does work. This is probably a bad way so I don't want to make a PR for it, but if someone comes across this PR, it might help them:

```
{
  "version": "0.2.0",
  "configurations": [
      {
          "type": "cppdbg",
          "program": "/usr/bin/systemctl",
          "request": "launch",
          "name": "systemd",
          "args": ["status", "systemd-resolved"],
          "cwd": "/root",
          "pipeTransport": {
              "pipeProgram": "mkosi",
              "pipeArgs": [
                  "-C",
                  "/path/to/project",
                  "ssh"
              ],
              "debuggerPath": "/usr/bin/gdb"
          },
          "MIMode": "gdb",
          "sourceFileMap": {
            "/root/build/../src": {
                "editorPath": "${workspaceFolder}",
                "useForBreakpoints": false
            },
            "/root/*": {
                "editorPath": "${workspaceFolder}/mkosi.builddir",
                "useForBreakpoints": false
            }
        }
      }
  ]
}

```

**Pull request merged**